oauth2_http/java/com/google/auth/oauth2/DownscopedCredentials.java

```diff
@@ -87,22 +88,32 @@
  */
 public final class DownscopedCredentials extends OAuth2Credentials {
 
-  private static final String TOKEN_EXCHANGE_ENDPOINT = "https://sts.googleapis.com/v1/token";
-
+  private final String TOKEN_EXCHANGE_URL_FORMAT = "https://sts.{universe_domain}/v1/token";
   private final GoogleCredentials sourceCredential;
   private final CredentialAccessBoundary credentialAccessBoundary;
+  private final String universeDomain;
+
   private final transient HttpTransportFactory transportFactory;
 
-  private DownscopedCredentials(
-      GoogleCredentials sourceCredential,
-      CredentialAccessBoundary credentialAccessBoundary,
-      HttpTransportFactory transportFactory) {
+  private final String tokenExchangeEndpoint;
+
+  /** Internal constructor. See {@link Builder}. */
+  private DownscopedCredentials(Builder builder) {
     this.transportFactory =
         firstNonNull(
-            transportFactory,
+            builder.transportFactory,
             getFromServiceLoader(HttpTransportFactory.class, OAuth2Utils.HTTP_TRANSPORT_FACTORY));
-    this.sourceCredential = checkNotNull(sourceCredential);
-    this.credentialAccessBoundary = checkNotNull(credentialAccessBoundary);
+    this.sourceCredential = checkNotNull(builder.sourceCredential);
+    this.credentialAccessBoundary = checkNotNull(builder.credentialAccessBoundary);
+
+    // Default to GDU when not supplied.
+    if (builder.universeDomain == null || builder.universeDomain.trim().isEmpty()) {
```

Comment:
nit: should we check the source credentials universe domain? order would be builder -> source credential -> default